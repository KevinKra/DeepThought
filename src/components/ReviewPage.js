import React from "react";
import NavBar from "./NavBar";
import "../styles/_ReviewPage.scss";
import ContentCard from "./ContentCard";
import Prompt from "./Prompt";

class ReviewPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      order: 0,
      hasBeenClicked: false,
      restartPrompt: false
    };
  }

  detectOrderNum = () => {
    console.log(this.state.order);
    if (this.state.order === 29) {
      this.setState({ restartPrompt: true });
    }
  };

  toMain = e => {
    e.preventDefault();
  };

  resetList = () => {};

  handleClick = e => {
    e.preventDefault();
    this.setState({
      hasBeenClicked: true
    });
  };

  resetList = () => {};

  nextQuestion() {}

  handleAnswerResponse = e => {
    e.preventDefault();
  };

  render() {
    if (!this.state.restartPrompt) {
    }
    return (
      <main className="window-reviewPage">
        <NavBar
          renderLoginPage={this.props.returnToLogin}
          renderMainPage={this.props.renderMainPage}
          mainLink={true}
        />
        <section className="content-section">
          {!this.state.restartPrompt ? (
            <React.Fragment>
              <ContentCard
                title="Question"
                handleClick={this.handleClick}
                // text={currentCard.question}
              />
              <ContentCard
                title="Answer"
                submitStatus={this.state.hasBeenClicked}
                handleAnswer={this.handleAnswerResponse}
                detectOrderNum={this.detectOrderNum}
                // text={currentCard.answer}
              />
            </React.Fragment>
          ) : (
            <Prompt
              restartPrompt={this.state.restartPrompt}
              toMain={this.toMain}
            />
          )}
        </section>
      </main>
    );
  }
}

export default ReviewPage;
