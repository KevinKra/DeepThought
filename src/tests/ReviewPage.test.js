import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ReviewPage from "../components/ReviewPage";

const mockCard = {
  id: 1,
  question: "What is JSX?",
  answer: "A JavaScript extension that imitates HTML",
  understood: false,
  reference: "https://reactjs.org/docs/introducing-jsx.html",
  timesPassed: 0,
  flagged: false
};

const mockFetchData = jest.fn();

const defaultState = {
  totalCards: [],
  failedCards: [],
  order: 0,
  hasBeenClicked: false,
  restartPrompt: false
};

describe("<ReviewPage />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ReviewPage />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
  // it("should update the totalCard array", () => {
  //   expect(wrapper.state().totalCards).toEqual([]);
  //   expect(mockFetchData).toBeCalled();
  // });
  // it("should display an answer card", () => {
  //   const wrapper = shallow(<ReviewPage />);
  //   expect(wrapper.instance().renderCard("answer")).toEqual(mockCard.answer);
  // });
});
