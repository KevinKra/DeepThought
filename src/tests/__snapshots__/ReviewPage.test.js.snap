// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewPage /> should match the snapshot 1`] = `
<main
  className="window-reviewPage"
>
  <NavBar
    mainLink={true}
  />
  <section
    className="content-section"
  >
    <ContentCard
      handleClick={[Function]}
      handleResponse={[Function]}
      renderCard={[Function]}
      title="Question"
    />
    <ContentCard
      handleResponse={[Function]}
      renderCard={[Function]}
      submitStatus={false}
      title="Answer"
    />
  </section>
</main>
`;
